sklearn/metrics/tests/test_dist_metrics.py

```diff
@@ -277,3 +287,25 @@ def test_readonly_kwargs():
     DistanceMetric.get_metric("seuclidean", V=weights)
     DistanceMetric.get_metric("wminkowski", p=1, w=weights)
     DistanceMetric.get_metric("mahalanobis", VI=VI)
+
+
+def test_minkowski_metric_validate_weights():
+    w1 = rng.random_sample(d)
+    w1[0] = -1337
+    msg = "w cannot contain negative weights"
+    with pytest.raises(ValueError, match=msg):
+        DistanceMetric.get_metric("minkowski", p=3, w=w1)
+
+    w2 = rng.random_sample(d + 1)
+    dm = DistanceMetric.get_metric("minkowski", p=3, w=w2)
+    msg = "size of w does not match"
+    with pytest.raises(ValueError, match=msg):
+        dm.pairwise(X1, X2)
+
+
+# TODO: Remove in 1.3 when mwinkowski is removed
+def test_wminkowski_deprecated():
+    w = rng.random_sample(d)
+    msg = "WMinkowskiDistance is deprecated in version 1.1"
+    with pytest.warns(FutureWarning, match=msg):
+        DistanceMetric.get_metric("wminkowski", p=3, w=w)
```

Comment:
Can you please check the equivalence with the weighted `MinkowskiDistance` with adjusted weights by `(w)**(1/p)`?